core/core.go

```diff
@@ -217,8 +218,9 @@ func New(config Config) (*App, error) {
 	})
 
 	// Add custom contract addresses if needed.
+	chainIDToContractAddresses := ethereum.NewChainIDToContractAddresses()
 	if config.CustomContractAddresses != "" {
-		if err := parseAndAddCustomContractAddresses(config.EthereumChainID, config.CustomContractAddresses); err != nil {
+		if err := parseAndAddCustomContractAddresses(config.EthereumChainID, chainIDToContractAddresses, config.CustomContractAddresses); err != nil {
```

Comment:
I think this is making things more complicated than they need to be. Instead of passing around `chainID` and `chainIDToContractAddresses`, why don't we just pass in `contractAddresses`? We don't need to keep looking up `chainIDToContractAddresses[chainID]` everywhere and there is no case I can think of where we would need to look at the contract addresses for more than one chain ID within a single `core.App` instance. This would save us memory, computation, and lines of code.

Reply:
Great, this was the feedback I was looking for on the draft. I wasn’t sure how flexible we wanted to leave this (a bunch of database callbacks actually read the chainID from orders, so i wondered if these could be different chainIDs from the app’s chainID in the future. It sounds like that isn’t the case)